### PR TITLE
feat: add A11yReport model

### PR DESCRIPTION
**Pull request by maxneuvians**
This PR adds the `A11yReport` model as defined in #9. It also adds better checks for JSONB fields in other models.

### CODE REVIEW

File: api/models/TemplateScan.py [hunk 1]
```diff
@@ -42,5 +42,5 @@ class TemplateScan(Base):
 
     @validates("data")
     def validate_data(self, _key, value):
-        assert value != ""
+        assert value is not None
```
Comment (dj2) [inline api/models/TemplateScan.py:45]: Does this still catch `""` ?

File: api/tests/models/test_A11yReport.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    assert a11y_report.revision == "revision"
+    assert a11y_report.url == "url"
+    assert_new_model_saved(a11y_report)
+    assert a11y_report.ci is False
+    assert a11y_report.summary == {"jsonb": "data"}
+    session.delete(a11y_report)
+    session.commit()
+
+
+def test_a11y_report_empty_productfails(scan_fixture, session):
```
Comment (dj2) [inline api/tests/models/test_A11yReport.py:61]: Similar version of these tests by `product=""` instead of just missing?

Comment (maxneuvians) [reply api/tests/models/test_A11yReport.py:61]: Yes those get caught:
```
    @validates("product")
    def validate_product(self, _key, value):
>       assert value != ""
E       AssertionError
```

File: api/models/TemplateScan.py (hunk 1 above)
Comment (maxneuvians) [reply api/models/TemplateScan.py:45]: I think that just becomes an empty json string, which is not deal either. I will add both tests

**Review by dj2 — APPROVED** [review]

**Pull request merged**